src/kem/frodokem/CMakeLists.txt

```diff
@@ -22,19 +24,20 @@ if(OQS_ENABLE_KEM_frodokem_1344_shake)
 endif()
 
 add_library(frodokem OBJECT ${SRCS})
+set(_FRODOKEM_OBJS ${_FRODOKEM_OBJS} $<TARGET_OBJECTS:frodokem>)
 
-if(OQS_USE_AVX2_INSTRUCTIONS)
-    if(OQS_USE_AES_INSTRUCTIONS)
-        if(OQS_ENABLE_KEM_frodokem_640_aes)
-            set(SRCS_AVX2 external/frodo640aes_avx2.c)
-        endif()
-        if(OQS_ENABLE_KEM_frodokem_976_aes)
-            set(SRCS_AVX2 ${SRCS_AVX2} external/frodo976aes_avx2.c)
-        endif()
-        if(OQS_ENABLE_KEM_frodokem_1344_aes)
-            set(SRCS_AVX2 ${SRCS_AVX2} external/frodo1344aes_avx2.c)
-        endif()
+set(SRCS_AVX2 "")
+if(OQS_PORTABLE_X86_64_BUILD OR OQS_USE_AVX2_INSTRUCTIONS)
```

Comment:
Same thing here: Why the dependency on OQS_PORTABLE_BUILD? Dependency on architecture is understood, but code presence IMO should otherwise (only) depend on OQS_USE_CPU_EXTENSIONS. @xvzcf : What's your take?

Reply:
No, OQS_USE_CPU_EXTENSIONS is not the right flag here.

The first thing we do with OQS_USE_CPU_EXTENSIONS is compile a program to test for the CPU features of the machine that we're compiling on (in .CMake/gcc_clang_intrinsics.cmake). We then set OQS_USE_AVX2_INSTRUCTIONS based on the output of that program.

The OQS_PORTABLE_BUILD flag enables runtime detection of CPU features and call dispatching. When it is enabled, we have to include this code whether or not the _compiling_ machine has AVX2 instructions. When it's not enabled, we include the code only if OQS_USE_AVX2_INSTRUCTIONS is set.

Reply:
I do not suggest adding OQS_USE_CPU_EXTENSIONS here -- that must have been set for OQS_USE_AVX2_INSTRUCTIONS even to be(come) active, right? I do propose checking for ARCH_X86_64 _and_ OQS_USE_AVX2_INSTRUCTIONS (independent of OQS_PORTABLE_BUILD, thus independent of OQS_PORTABLE_X86_64_BUILD).

Reply:
OK. I see your point w.r.t. OQS_USE_AES_INSTRUCTIONS, but not for OQS_USE_AVX2_INSTRUCTIONS.

The name "AVX2" refers to an X86 extension, so checking ARCH_X86_64 is redundant---we will never set OQS_USE_AVX2_INSTRUCTIONS if the architecture is not X86_64.

For AES I think it would be better to rename the flag "OQS_USE_AES_NI_INSTRUCTIONS", as "AES-NI" unambiguously refers to the X86 extension. If we ever implement a compile-time check for ARM CPU features we should use a different name like "OQS_USE_ARMV8_AES_INSTRUCTIONS".

For this PR, I think it's fine for "OQS_USE_AES_INSTRUCTIONS" to refer to AES-NI for now.